docs/proposal-rose-suite-run.md

```diff
@@ -39,66 +39,234 @@ The remaining points can happen any time after 1) is complete although note that
      when performing operations on a suite?
 
-8. Consider the behaviour created by the `CYLC_VERSION` and `ROSE_VERSION`
-   variables in `rose-suite.conf`. These allow rose suite-run to:
-   * Change the version of Cylc and check the version of Rose used to run the
-     suite.
-   * Prevent suites accidentally upgrading or downgrading versions on restart.
-
-9. Consider `rose prune` functionality.
+8. Consider `rose prune` functionality.
    * Need to make sure it will continue to work correctly.
-   * Need Cylc equivalent but this should be done as part of
+   * Longer term we should build housekeeping into Cylc, see
      [cylc-flow #1159](https://github.com/cylc/cylc-flow/issues/1159).
 
-### CLI
+9. Record version control information when you install a suite, see
+   [cylc-flow #3849](https://github.com/cylc/cylc-flow/issues/3849).
+   * `rose suite-run` current supports this for Subversion only
+     (`svn info` + `svn status` + `svn diff`).
+   * Extend this to support equivalent information for git (and preferably make
+     it easy to plugin support for other systems as well).
+
+### CLI changes
+
+`cylc install` - new command to install workflows.
+* By default, `cylc install` will install the workflow found in `$PWD` into
+  `~/cylc-run/$(basename $PWD)/runN` (where `runN` = `run1`, `run2` ...).
+* If `--run-name` is not specified and `run1` already exists it will install
+  into `run2` (and so on).
+  Create a symlink `runN` pointing at the latest run.
+* `--run-name=my-run` implies install into `~/cylc-run/$(basename $PWD)/my-run`.
+  If the target directory already exists then fail.
+* `--no-run-name` implies install into `~/cylc-run/$(basename $PWD)`.
+* `--flow-name=my-flow` implies install into `~/cylc-run/my-flow/runN`.
+* `--directory=/path/to/flow` (`-C ...`) implies install the workflow found in
+  `/path/to/flow` (rather than `$PWD`).
+* Installation will involved copying over the files found in the source
+  directory.
+  * This will exclude any `.git` or `.svn` directories.
+    Everything else will be copied.
+    * If needed we could make this configurable via a `.cylcignore` file (future
+      enhancement).
+  * The installation should fail if `log`, share` or `work` exist in the source
+    directory.
+  * The installation should fail if neither `suite.rc` nor `cylc.flow` exist.
+  * The installation should fail if the target directory is not valid.
+    e.g. you cannot install into `~/cylc-run/my-flow/runN` if `~/cylc-run/my-flow`
+    already contains an installed suite.
+* Version control information will be recorded where relevant (svn/git).
+* The `log, `share`, share/cycle` and `work` directories  will be created
+  following whatever symlink rules are defined for localhost
+  (over-ridable via command line options).
+  * This will continue to support "symlinked alternate run directories"
+    (see [cylc-flow #2935](https://github.com/cylc/cylc-flow/pull/2935)).
```

Comment:
Missed this on my first scan, the symlink functionality will allow the entire run dir to be symlinked to another location which would effectively replace this functionality?

```suggestion
  * This will replace "symlinked alternate run directories" functionalty
    (see [cylc-flow #2935](https://github.com/cylc/cylc-flow/pull/2935)).
```

May be desirable to implement a replacement CLI option though?

Reply:
Change applied.
I've haven't tried to  define what command line options will be used - we can decide this later.